### ISSUE TITLE
[Bug]: trimScale on 0 value results in an infinite loop.

### ISSUE THREAD

**Issue by pnappa** [dinerojs/dinero.js#640]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When supplying a zero-y value to trimScale, it will never halt.

### Expected behavior

Return a dinero object with properties: `{ amount: 0, scale: 2 }` (as `2` is the currency's exponent in this case).

### Steps to reproduce

Minimum reproducer:
```
trimScale(dinero({ amount: 0, currency: _.USD }));
```

### Version

2.0.0-alpha.8

### Environment

Firefox (can reproduce in the devtools library at v2.dinerojs.com), or NodeJS

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by svarlet** [comment]
I observed the same problem. It happens in my jest unit tests, and I am able to reproduce it in the codesandbox.io that you kindly provide.

I found out when I tried to allocate one USD cent equally into 3 buckets:
```typescript
const oneCent = dinero({amount: 1, currency: USD, scale: 2});
const allocate(oneCent, [1, 1, 1]); // allocated as: [1, 0, 0]
allocate.map(trimScale); // infinite loop of death?
```

**Comment by svarlet** [comment]
Furthermore, I noticed that this bug is already fixed by https://github.com/dinerojs/dinero.js/pull/448 but it was merged after the alpha-8 release. Am I mistaken if I think a new release would address this problem?

**Comment by sarahdayan** [comment]
Hey, this is fixed in [v2.0.0-alpha.9](https://github.com/dinerojs/dinero.js/releases/tag/v2.0.0-alpha.9).
